src/Headers/Ic.h

```diff
@@ -104,7 +105,7 @@ class Ic
   virtual void Generate(void) {
     ExceptionHandler::getIstance().raise("Ic::Generate function not implemented");
   };
-  virtual FLOAT GetDensity(const FLOAT *) const {
+  virtual FLOAT GetDensity(const FLOAT *, const Typemask typemask=Typemask(true)) const {
```

Comment:
I believe this should take a const Typemask&, rather than a const Typemask.

Are we sure that using a Typemask is the right way to go here: It's definitely the most flexible so I won't argue on that ground, but it is a bit of a pain to use when we want just a single particle type. The other option would be to make GetDensity take a ptype. We'd then just call it multiple times if we wanted a 'total density' over some set of particles.

By the way, these functions should not be virtual.  Also, I think they can be declared inline. If so, I'd argue that they should be inline, this would allow the compiler to do much more optimization (particularly when looping over types if we cange the interface).